src/process.ts

```diff
@@ -174,20 +174,36 @@ export class PowerShellProcess {
         return true;
     }
 
-    private waitForSessionFile(): Promise<utils.IEditorServicesSessionDetails> {
-        return new Promise((resolve, reject) => {
-            utils.waitForSessionFile(this.sessionFilePath, (sessionDetails, error) => {
-                utils.deleteSessionFile(this.sessionFilePath);
+    private sleep(ms: number) {
+        return new Promise(resolve => setTimeout(resolve, ms));
+    }
+
+    private async waitForSessionFile(): Promise<utils.IEditorServicesSessionDetails> {
+        const numOfTries = this.sessionSettings.developer.waitForSessionFileNumOfTries;
 
-                if (error) {
-                    this.log.write(`Error occurred retrieving session file:\n${error}`);
-                    return reject(error);
-                }
+        // This is used to warn the user that the extension is taking longer than expected to startup.
+        const warnUserThreshold = numOfTries / 2;
```

Comment:
For the warning, I'd imagine our standard expectation for startup time is a constant, like 30 seconds, rather than parameterised by the setting